test: add registerTemplate

otaviopace opened the pull request:
This commit adds multiple tests for the
registerTemplate function of the lib/register/template.js
file.

*Issue #, if available:*
This closes https://github.com/amzn/style-dictionary/issues/142

*Description of changes:*
Creation of tests about `lib/register/template.js`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

otaviopace commented:
I don't know. If that is correct, I can change it :slightly_smiling_face:

dbanksdesign commented:
Yes, please use the 'jest' branch. This is part of a larger project to migrate our tests to Jest and generally improve the testing of this package. You will probably have to make some minor changes so it works with Jest instead of Mocha. Also if you could remove the package-lock.json from the PR that would be great because you aren't changing any dependencies with this PR.

Thanks for stepping up on this, we really appreciate it!

otaviopace commented:
Done! I've rebased my branch with the `jest` branch and changed the Pull Request to be correct :slightly_smiling_face: 
Wierdly the tests started to fail, do you guys know if the `jest` branch is passing on CI?

dbanksdesign commented:
Could you also take a look at the directory/file structure of the tests in the jest branch? I don't think this test is actually running, I can't see it on the CI build.

otaviopace commented:
Since #148 was merged, I rebased my branch with the fixed `jest` and removed the `package-lock.json` from the `commit`. Can you check again if we can merge @dbanksdesign ? :slightly_smiling_face:

dbanksdesign commented:
The test looks good, but could you try rebasing from the 'jest' branch again? The reason I asked is in the jest branch the tests were moved from `test/` to `__tests__/`

dbanksdesign commented:
Err I guess not rebase again, but just move the test to the proper directory. Sorry!

otaviopace commented:
Can this be merged @dbanksdesign, now that it is fixed :slightly_smiling_face:  ?

otaviopace commented:
Horray!!! :tada:
